### PR TITLE
feat: add /how-to/detect-ipfs-on-web

### PR DESCRIPTION
**Pull request by lidel**
This PR:

- tweaks mermaid diagrams config to match IPFS brand book color guide and render a readable size, even on mobile
-  adds `/how-to/detect-ipfs-on-web` page with some best practices around detecting IPFS resources on the web, based on recent discussions with Brave team and our effort to streamline implementation there.
   - Based on my experience with IPFS Companion browser extension and Brave,  I aimed at a healthy balance between low level details and end user experience, but lmk if this is hard to follow or could be improved.
   - cc @cypt4 @autonome @SgtPooki  @meandavejustice  @rangermauve @fabricedesre @juliaxbow for visibility and additional feedback

As per usual, apologies for any grammar errors ;-)

## PR  preview

 👉️  https://github.com/ipfs/ipfs-docs/blob/feat/how-to-implement-support-in-a-web-browser.md/docs/how-to/detect-ipfs-on-web.md
 
 ## Website preview  (static)
 
👉️  https://bafybeicp2oota2okffb4u6ta6gzx46jnpyyzxdrrzpzivtbo4xv6sfefim.on.fleek.co/how-to/detect-ipfs-on-web/

To see how the final docs.ipfs.tech version would look, see Fleek preview link in the CI section below.

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

File: docs/how-to/detect-ipfs-on-web.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Detect IPFS on the Web
```
Comment (autonome) [inline docs/how-to/detect-ipfs-on-web.md:1]: Title should probably be more general, because is not just about detection. eg "Handling IPFS Content in Web User Agents" or something in that direction.

File: docs/how-to/detect-ipfs-on-web.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+The high level heuristic is to check, in order:
+- Does the `URL` follow [path or subdomain gateway](/how-to/address-ipfs-on-web/) convention?
+- If not, does [DNSLink](/concepts/dnslink/) with an IPFS path exist for `URL.hostname`?
+
+If any of the above is true, an user agent should be able to find a content path that can be used for data retrieval over IPFS. See the illustration below.
+
+```mermaid
+flowchart TD
+    Start[Top-level navigation event]
```
Comment (autonome) [inline docs/how-to/detect-ipfs-on-web.md:19]: If diagram only handles top-level navigation events, should add a note because policy there is typically different than other load points. Maybe point also to the test page.

**Review by autonome — APPROVED** [review]

File: docs/how-to/detect-ipfs-on-web.md (hunk 2 above)
Comment (lidel) [reply docs/how-to/detect-ipfs-on-web.md:19]: The path detection diagram is generic enough to be applied to subresource requests too – the usual limitations we experience are vendor specific (example: https://github.com/ipfs/ipfs-companion/issues/1095).

TODO(note to self): think how to convey that. Probably change labels on diagram to be about all requests + add section explaining  caveats around "top-level navigaton" and "subresource requests" on HTTP and `ipfs://`.

File: docs/how-to/detect-ipfs-on-web.md (hunk 1 above)
Comment (lidel) [reply docs/how-to/detect-ipfs-on-web.md:1]: We already have "Addressing IPFS on the Web" so I went with similar title, but not feeling strong either way.

**Comment by ElPaisano** [comment]
@lidel @SgtPooki @autonome is this PR still needed? Cleaning out some old PRs now.

**Comment by BigLep** [comment]
+1 on merging so we have something, and the moving to specs when it's ready to house it.  Can you tackle this please @ElPaisano ?

**Comment by aschmahmann** [comment]
from @lidel: This is fine to merge, but also ok to keep as a draft until we have time to move this to the specs repo.

File: docs/how-to/detect-ipfs-on-web.md (hunk 1 above)
Comment (ElPaisano) [inline docs/how-to/detect-ipfs-on-web.md:1]:
```suggestion
# Discover IPFS content from web2
```

File: docs/how-to/detect-ipfs-on-web.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+# Detect IPFS on the Web
+
+Below is the suggested heuristic for detecting content-addressed resources on the web, and presenting additional user interface for interacting with them.
```
Comment (ElPaisano) [inline docs/how-to/detect-ipfs-on-web.md:3]:
```suggestion
Below is the suggested heuristic for detecting content-addressed resources on the web, and presenting additional user interface for interacting with that content.
```

File: docs/how-to/detect-ipfs-on-web.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+The goal of this page is to provide some suggestions and best practices for web browsers and other [user agents](https://en.wikipedia.org/wiki/User_agent) that are interested in adding support for [content-addressed resources on the Web](/how-to/address-ipfs-on-web).
```
Comment (ElPaisano) [inline docs/how-to/detect-ipfs-on-web.md:5]:
```suggestion
The goal of this page is to provide some suggestions and best practices for web2 web browsers and other [user agents](https://en.wikipedia.org/wiki/User_agent) that are interested in adding support for IPFS [content-addressed resources on the Web](/how-to/address-ipfs-on-web).
```

**Comment by ElPaisano** [comment]
@lidel this is ready for review.  Made some adjustments

Main thing that is not clear to me is this section https://bafybeiblvih4yq6cbkxd7tauokc6qaxvvsvraftj7wa6sza6qyt5ggdbha.on.fleek.co/how-to/detect-ipfs-on-web/#what-to-do-with-detected-paths (starting on line 88)

Given that "The goal of this page is to provide some suggestions and best practices for Web 2 web browsers and other [user agents](https://en.wikipedia.org/wiki/User_agent) that are interested in adding support for [content-addressed resources](./address-ipfs-on-web.md).", do you mean that any browsers supporting IPFS should implement an "Open via IPFS" button that displays one of the various path types listed? I think SgtPooki had similar comments.

 Also, https://github.com/ipfs/ipfs-docs/pull/1295#discussion_r1019539221 is still not clear to me
